src/modelmanager.cpp

```diff
@@ -150,6 +150,19 @@ void processNodeInputs(const std::string nodeName, const rapidjson::Value::Const
     }
 }
 
+void processPipelineInputs(const rapidjson::Value::ConstMemberIterator& pipelineInputsPtr, const std::string& nodeName, std::unordered_map<std::string, std::string>& nodeOutputNameAlias, const std::string& pipelineName) {
+    for (const auto& pipelineInput : pipelineInputsPtr->value.GetArray()) {
+        const std::string pipelineInputName = pipelineInput.GetString();
+        SPDLOG_INFO("Alliasing node:{} output:{}, under alias:{}",
+            nodeName, pipelineInputName, pipelineInputName);
+        if (nodeOutputNameAlias.count(pipelineInputName) > 0) {
+            SPDLOG_WARN("Pipeline {} has duplicated input declaration", pipelineName);
+        } else {
+            nodeOutputNameAlias[pipelineInputName] = pipelineInputName;
+        }
```

Comment:
minor:
Efficient construction for this would be:
auto it = nodeOutputNameAlias.find(name)
if (it == nodeOutputNameAlias.end()) {
   SPDLOG_ERROR
} else {
    (*it).second = name;
}

WIth current way you are doing search on a map twice.

Reply:
The logic here is opposite: we only insert if we cannot find key. Nevertheless we can switch to use .find()

Reply:
Done, I used .insert(), which returns information if entry was added or skipped due to key duplicatio.